src/phoenix/experimental/evals/templates/template.py

```diff
@@ -15,19 +14,27 @@
 NOT_PARSABLE = "NOT_PARSABLE"
 
 
-@dataclass
 class PromptTemplate:
     text: str
     variables: List[str]
 
-    def __init__(self, text: str, delimiters: List[str] = [DEFAULT_START_DELIM, DEFAULT_END_DELIM]):
+    def __init__(
+        self,
+        text: str,
+        delimiters: List[str] = [DEFAULT_START_DELIM, DEFAULT_END_DELIM],
+    ):
         self.text = text
         self._start_delim, self._end_delim = self._get_delimiters(delimiters)
-        self._validate()
-        self.variables = self._parse_variables()
+        self._validate(self.text)
+        self.variables = self._parse_variables(self.text)
 
-    def format(self, variable_values: Dict[str, Union[bool, int, float, str]]) -> str:
-        prompt = self.text
+    def prompt(self, **options: Any) -> str:
+        return self.text
```

Comment:
I'm not sure this function is doing anything useful?

Reply:
The idea was to keep the interface for interacting with "PromptTemplate"-like interfaces consistent, there's kind of an implied abstract class here that I haven't extracted for the sake of the example code here, but this method is extended for the `ClassificationTemplate` object, allowing us to avoid managing branching codepaths in `llm_classify`